src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -3210,7 +3212,10 @@ JvmtiEnv::GetMethodName(Method* method, char** name_ptr, char** signature_ptr, c
 jvmtiError
 JvmtiEnv::GetMethodDeclaringClass(Method* method, jclass* declaring_class_ptr) {
   NULL_CHECK(method, JVMTI_ERROR_INVALID_METHODID);
-  (*declaring_class_ptr) = get_jni_class_non_null(method->method_holder());
+  Klass* k = method->method_holder();
+  Handle holder(Thread::current(), k->klass_holder()); // keep the klass alive
+  // Cannot check klass_holder == nullptr because klass could have null loader holder
+  (*declaring_class_ptr) = k->is_loader_alive() ? get_jni_class_non_null(k) : nullptr;
```

Comment:
This condition is already checked by the caller, so the !k->is_loader_alive() case should be dead code. I would prefer an assert to check that surely it is alive or we shouldn't be here.